sdk/core/core-http/lib/util/xml.browser.ts

```diff
@@ -147,3 +152,25 @@ function buildNode(obj: any, elementName: string): Node[] {
     throw new Error(`Illegal value passed to buildObject: ${obj}`);
   }
 }
+
+export function convertAtomXmlToJson(body: string): any {
+  const dom = parser.parseFromString(body, "text/xml");
```

Comment:
The difference between `convertAtomXmlToJson()` and `parseXML()` seems to be in the mime type being passed to `parser.parseFromString()`.
One uses `application/xml`, the other uses `text/xml`.

As per [documentation on parseFromString()](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser), the mime type is used to control the type of the return value.
Both `application/xml` and `text/xml` result in the type of the return value being `XMLDocument`.

Therefore, why can we not use `parseXML()` directly?

Reply:
We talked about this offline.
- My proposal was that Atom XML specific things should live in core-http and not the consumer SDKs. But we decided on taking this up as an improvement later. As such, it makes sense to retain the differnt utilities.
- Moreover if we have convertToAtomXML defined, it would be more organized to have its counterpart as well - this sits inline with the right long term design we agreed on.
- `application/xml` is something that's being specifically set and is a content type referred to in deserializationPolicy as well and might be what the service expects as well.

Reply:
Another reason why we can't reuse parseXml() is about the parser and builder settings being different for across the operations as discussed